check_binary.sh

```diff
@@ -360,3 +360,16 @@ if [[ "$DESIRED_CUDA" != 'cpu' && "$DESIRED_CUDA" != *"rocm"* ]]; then
     popd
   fi # if libtorch
 fi # if cuda
+
+###############################################################################
+# Check PyTorch supports TCP_TLS gloo transport
+###############################################################################
+
+if [[ "$(uname)" == 'Linux' ]]; then
+  GLOO_DEVICE_TRANSPORT=TCP_TLS MASTER_ADDR=localhost MASTER_PORT=63945 python -c "import torch; import torch.distributed as dist; print(torch.__version__); dist.init_process_group('gloo', rank=0, world_size=1)" | grep "unsupported gloo device" &> /dev/null
+  RESULT=$?
+  if [ $RESULT -eq 0 ]; then
```

Comment:
We should check that result is not equal to zero, is it?
```suggestion
  if [ $RESULT -ne 0 ]; then
```

Reply:
I don't think so: https://man7.org/linux/man-pages/man1/grep.1p.html

       The following exit values shall be returned:

        0    One or more lines were selected.

        1    No lines were selected.

       >1    An error occurred.